### PR TITLE
feat: set lock

### PR DESCRIPTION
**Pull request by Ad96el**

### CODE REVIEW

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 1]
```diff
@@ -304,7 +316,7 @@ pub mod pallet {
 			Ok(())
 		}
 
-		#[pallet::call_index(2)]
+		#[pallet::call_index(1)]
```
[the remark this answers is not in the document]
Comment (ntn-x2) [reply pallets/pallet-bonded-coins/src/lib.rs:319]: Why not 0?

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 2]
```diff
@@ -193,6 +200,11 @@ pub mod pallet {
 		Destroyed {
 			id: T::PoolId,
 		},
+		/// The manager of a pool has been updated.
+		ManagerUpdate {
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs]: [00bec83](https://github.com/KILTprotocol/kilt-node/pull/773/commits/00bec83d6107f67120bda4a9efa33800561f6501)

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs:319]: Yeah, the call indices were a bit odd, prob happened during the development and adding more and more tx.

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 3]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
+			Pools::<T>::try_mutate(&pool_id, |pool| -> DispatchResult {
+				let entry = pool.as_mut().ok_or(Error::<T>::PoolUnknown)?;
+				ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
+
+				entry.state = PoolStatus::Locked(lock.clone());
+
+				Self::deposit_event(Event::LockSet {
+					id: pool_id.clone(),
+					lock,
+				});
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs]: :cry: [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 4]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
+			Pools::<T>::try_mutate(&pool_id, |maybe_entry| -> DispatchResult {
 				let entry = maybe_entry.as_mut().ok_or(Error::<T>::PoolUnknown)?;
 				ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
-				entry.manager = manager;
+				entry.manager = manager.clone();
+
+				Self::deposit_event(Event::ManagerUpdate {
+					id: pool_id.clone(),
+					manager,
+				});
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs:367]: 😢 [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 5]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
 
 		#[pallet::call_index(4)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn unlock(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			Pools::<T>::try_mutate(&pool_id, |pool| -> DispatchResult {
+				let entry = pool.as_mut().ok_or(Error::<T>::PoolUnknown)?;
+				ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
+				entry.state = PoolStatus::Active;
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs:402]: Good point: [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 6]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
+
+				Self::deposit_event(Event::Unlocked { id: pool_id.clone() });
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs]: 😢 [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 7]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
+				});
+
 				Ok(())
 			})
 		}
 
 		#[pallet::call_index(4)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn unlock(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs]: [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 8]
```diff
@@ -336,24 +348,66 @@ pub mod pallet {
...
+				});
+
+				Ok(())
+			})
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn set_lock(origin: OriginFor<T>, pool_id: T::PoolId, lock: Locks) -> DispatchResult {
+			let who = ensure_signed(origin)?;
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs]: 😢 [92eaf15](https://github.com/KILTprotocol/kilt-node/pull/773/commits/92eaf153247231c0925b3c47682385b37929dd73)

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/lib.rs:367]: this hasn't changed though?

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 9]
```diff
@@ -336,24 +348,69 @@ pub mod pallet {
...
+				});
+
 				Ok(())
 			})
 		}
 
+		#[pallet::call_index(3)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn set_lock(origin: OriginFor<T>, pool_id: T::PoolId, lock: Locks) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
```
Comment (rflechtner) [inline pallets/pallet-bonded-coins/src/lib.rs:376]: we prob have to have another look at the origins, now some of the management origins are PoolCreateOrigins, others are DefaultOrigins?

**Pull request merged**